TreeList - Getting Started: Add ASP.NET Core Overhaul (#8970)
DevExpress/devextreme-documentation: concepts/05 UI Components/TreeList/00 Getting Started with TreeList/40 Edit and Validate Data.md

        </DxTreeList>
    </template>

    <script setup lang="ts">
    import { DxTreeList, DxEditing, DxColumn, DxRequiredRule } from 'devextreme-vue/tree-list';

    </script>

##### React

    <!-- tab: App.tsx -->
    import { TreeList, Editing, Column, RequiredRule } from 'devextreme-react/tree-list';

    function App() {
        return (
            <TreeList>
                <Editing
                    mode="popup"
                    allowUpdating={true}
                    allowDeleting={true}
                    allowAdding={true}
                />
                <Column dataField="FullName">
                    <RequiredRule />
                </Column>
                <Column dataField="Position">
                    <RequiredRule />
                </Column>
                <Column dataField="BirthDate">
                    <RequiredRule />
                </Column>
                <Column dataField="HireDate">
                    <RequiredRule />
                </Column>
                <Column dataField="State">
                    <RequiredRule />
                </Column>
                {/* ... */}
            </TreeList>
        );
    }

---

[note]

> Hierarchical data sources do not support editing in the TreeList component out of the box. To configure editing operations in hierarchical data sources, refer to the following example:

Reviewer: ```suggestion
The TreeList component does not support editing hierarchical data sources out of the box. To configure editing operations for hierarchical data sources, refer to the following example:
```
